src/ontology/mondo.Makefile

```diff
@@ -268,8 +268,55 @@ reports/%-rare-diseases.tsv: $(ONT)-base.owl reports/%-rare-diseases.txt
 rare-disease-reports: reports/old-rare-diseases.tsv reports/new-rare-diseases.tsv
 	python ../scripts/filter_rare_disease_list.py reports/old-rare-diseases.tsv reports/new-rare-diseases.tsv reports/added-rare-disases.tsv reports/removed-rare-diseases.tsv
 
-	
+#############################################
+##### One-time scripts ######################
+#############################################
+# MedGen conflicts (Aug 2023) pipeline
```

Comment:
_Nico wrote:_
- `xref: GTR:AN1112503 {source="UMLS:CN706304"}` is removed, but should not. Remember #6573! This PR needs to be merged first before your pipeline is run!
- You remove stuff like 
    - `+def: "Deficient blood distribution to the limbs caused by narrowing or obstruction of the lumen of the peripheral arteries." [MEDGEN:115991]`
    - `+synonym: "XYLT1-CDG - xylosyltransferase 1 congenital disorder of glycosylation" EXACT [MEDGEN:C4750849]`
    - Please do not remove these! only `^xref:` matches!
- If a row contains one of the following strings, do not remove it: `MONDO:preferredExternal`, `https://orcid.org` (indicates someone has manually removed it)
- Question: for all the UMLS CUIs we remove, will we get one back from the pipeline?

Reply:
Hey @matentzn, good review. I copy/pasted your comment here because I think, to reduce clutter and ensure everything is completed, it's a good idea to make as many comments possible from the "files changed" tab, so that that we get the "resolve conversation" button. I also enumerated your points.

Reply:
Re 1 and 2: I guess we should have clarified that "remove xrefs" means: if the xref appears in an axiom annotation (in wavy or squared brackets) remove the axiom annotation with the xref, not the whole row in the OBO file, if the xref appears in a term level `xref: ...` statement, remove the whole row. 

Thanks for the clarifications on 4, I ready your comments. I trust you to basically have an idea of what you are doing now, and that removing the 2/5 means removing _wrong_ xrefs. I don't have enough time to ensure that these 2/5 are considered wrong by Megan; I will trust it is. So I am fine as you are doing it then!

Reply:
@matentzn Today I made fixes / updates regarding topics 1-3 above.

**Rebased after your #6573**
Regarding topic 1 above, this might have been an oversight/confusion on my part. I've left the `UMLS:CN` removals to your PR. I reset my changes to `mondo-edit.obo` and rebased.

**MEDGEN removals: undid non-xrefs**
Regarding topic 2 above, that feels a bit dumb on my part to have removed every line if "MEDGEN:" simply appeared in it; I had an inkling you might have meant to remove MEDGEN xrefs lines and source annotations separately.

See code snippet below for how this is now implemented. For MEDGEN xrefs, the whole lines are removed. I then was going to implement the logic for situations like `source="MEDGEN:`. However, there was only 1 such case and it was already being removed because it was one of Megan's conflicts.

**Python reimplementation**
Because of the nature of topic 3 above (exceptions), I re-implemented the whole make/ShellScript method of doing these removals (`grep -v`) to pure Python. Also seems lot easier / more straightforward in Python, and greatly increases accuracy.

https://github.com/monarch-initiative/mondo/blob/9fd9171bad755556d93b1e34ee32e46d9211c478/src/scripts/medgen_conflicts_removals.py#L82-L94

<details><summary>Updated metrics table</summary>
<p>

These results look good. `n_single_cui_removed_from_xrefs_n_times__1` is 2,818 this time; was previously 2,819. I' guessing this is because I rebased from your recent PR, which must have removed 1 of these already.

I didn't add an entry to this table for `n_medgen_source_evidence_xref_annotations_removed`. But there were 0 anyway. All of these ones that would have been removed this way ended up being removed as a consequence of Megan's UMLS: xref removals anyway.

|metric_name                               |val |
|------------------------------------------|----|
|n_lines_removed                           |3086|
|n_xrefs_removed                           |3086|
|n_non_xrefs_removals                      |0   |
|n_medgen_removals                         |28  |
|n_umls_removals                           |3058|
|n_single_cui_removed_from_xrefs_n_times__1|2818|
|n_single_cui_removed_from_xrefs_n_times__2|106 |
|n_single_cui_removed_from_xrefs_n_times__3|8   |
|n_single_cui_removed_from_xrefs_n_times__4|1   |


</p>
</details> 

**Can increase specificity of removals, if you want**
You said that it was alright for any CUIs / UMLS IDs Megan mentioned in her conflicts file, that I can remove them even if they end up getting removed in multiple places. However, she did mention specific classes where this was a problem. Now that I have changed `grep -v` --> pure Python, I can change it to remove only the xrefs on these classes. If I do this, I can re-run the analysis table above to show that all CUIs were only removed from 1 xref.

**Removal exceptions MONDO:preferredExternal & https://orcid.org**
You said:
> If a row contains one of the following strings, do not remove it: MONDO:preferredExternal, https://orcid.org (indicates someone has manually removed it)

I'm only doing 3 kinds of deletes. Here they are, as well as a count of how many times I found either of these strings:
1. source="MEDGEN: - 0
2. xref: MEDGEN: - 0
3. Medgan's "xref: UMLS:" conflicts - 23

So, when it comes to my removals of anything MEDGEN/UMLS related, the only time I'm finding these strings (in the xref axiom annotations) is for cases where Megan specifically asked me to remove an xref.

However, I take it that you do not actually want me to override her, so (as can be seen in my Python snippet above), I did not implement this exception logic for those conflicts.

Also, just to clarify--what do you mean by "it" in "indicates someone has manually removed it" in your quote above? I think it means "UMLS or MEDGEN source= evidence axiom annotations". Is that correct?

<details><summary>Examples of such rows I found</summary>
<p>

1. `xref: UMLS:C1970879 {source="MONDO:equivalentTo", source="MONDO:ncbi_mim2gene_medline", source="MONDO:preferredExternal", source="OMIM:300636"}`
2. `is_a: MONDO:0000762 {source="DC-OMIM:300869", source="https://orcid.org/0000-0002-4142-7153"} ! syndrome caused by partial chromosomal duplication`

</p>
</details>

Reply:
This looks more or less like I would expect now. I have gathered the 900 more risky removals in this spreadsheet: https://docs.google.com/spreadsheets/d/1y86gfXydQE0wOfQzddohN-TWWHUJ11QH7IJg9n_PA5c/edit#gid=0

I need to be convinced that all of these are really wrong. Can you tell me, according to the data (I know you told me before, but I needed to ask the question again):

**Does Megan consider all of these wrong one way or another?**

I am still not sure if we should remove UMLS xrefs only from specific Mondo IDs, or from all of them. Maybe now it is time to organise a call with Megan?

Let me know when you have the ROBOT template together, this will give me some important clues to advice you better. Thanks!!

Reply:
Glad this is better. Yes, Megan considers all of these problematic and should be removed some way in Mondo. As you understand, for many of them she is only pointing out that they need to be removed on a single class. But I will continue removing all of their xrefs for now.

Perhaps the best way to see what is recommended for each sheet is in my code comments:
https://github.com/monarch-initiative/mondo/blob/9fd9171bad755556d93b1e34ee32e46d9211c478/src/scripts/medgen_conflicts_removals.py#L31-L61

This is a distillation of what I've written on each sheet in my personal notes, which is basically a distillation of what Megan has written in [README_July2023_CUIReports_FromMedGentoMondo.txt](https://github.com/monarch-initiative/mondo-ingest/files/12029711/README_July2023_CUIReports_FromMedGentoMondo.txt) ([her original comment](https://github.com/monarch-initiative/mondo-ingest/issues/273#issuecomment-1632774012)).

Feel free to schedule a call if you want. Personally what she's asking seems pretty clear to me; I don't feel the need for any clarification. But perhaps you feel that need, or maybe we could evaluate reasoning / ask other questions that you may have. Below is my schedule for the next two weeks.

<details><summary>Joe's availability</summary>
<p>

<img width="1506" alt="Screen Shot 2023-08-27 at 4 37 50 PM" src="https://github.com/monarch-initiative/mondo/assets/13045020/0c1bfb62-167b-48b3-9099-266246a90ac4">

<img width="1513" alt="Screen Shot 2023-08-27 at 4 37 59 PM" src="https://github.com/monarch-initiative/mondo/assets/13045020/5f7337b2-2d94-4672-9478-8b82b22adf5f">


</p>
</details>

Reply:
@matentzn Marking this one resolved. I think the only important one left here is my last comment. You were just feeling unsure of some of these removals and maybe wanted to meet with Megan. I explained the removals above; I don't really see a need to meet with her at this time.